### PR TITLE
Add service attributes

### PR DESCRIPTION
**Pull request by EddyK69**
## Type of change
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to this library)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

Added service attributes to make it easer In HA-integration to loop over the available attributes per service.
(After this merge, the ATTR_LAST_TRIP, ATTR_ALL_TRIPS & ATTR_CHARGING_PROFILE can be removed from the sensor.py in HA-integration...
 
- This PR is related to issue: https://github.com/bimmerconnected/ha_custom_component/pull/5

## Checklist

- [X] The code change is tested and works locally.
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

File: bimmer_connected/all_trips.py [hunk 1]
```diff
@@ -101,6 +101,16 @@ def __getattr__(self, item):
...
         return self._state.attributes[SERVICE_ALL_TRIPS][item]
 
+    @property
+    def available_attributes(self) -> List[str]:
+        """Get the list of all-trips attributes available for this vehicle."""
+        result = ['average_combined_consumption', 'average_electric_consumption',
+            'average_recuperation', 'battery_size_max', 'chargecycle_range',
+            'reset_date', 'saved_co2', 'saved_co2_green_energy', 
+            'total_electric_distance', 'total_saved_fuel'
+        ]
```
Comment (gerard33) [inline bimmer_connected/all_trips.py:111]: Are all these attributes allways available for a car which has `all_trips`?
Same question of course for the `charging_profile` and `last_trips`?

File: bimmer_connected/all_trips.py [hunk 2]
```diff
@@ -101,6 +101,16 @@ def __getattr__(self, item):
         """Generic get function for all backend attributes."""
         return self._state.attributes[SERVICE_ALL_TRIPS][item]
 
+    @property
+    def available_attributes(self) -> List[str]:
```
Comment (gerard33) [inline bimmer_connected/all_trips.py:105]: The test fail due to List.

```
def available_attributes(self) -> List[str]:
E   NameError: name 'List' is not defined
```

**Comment by gerard33** [comment]
Can you have a look at the failing CI checks?
https://travis-ci.com/github/bimmerconnected/bimmer_connected/builds/201901979

File: bimmer_connected/all_trips.py (hunk 1 above)
Comment (EddyK69) [reply bimmer_connected/all_trips.py]: Yep, as far as I know yes. I only have an I3 and mine has; and I checked the responses in the test-folder also...

**Comment by EddyK69** [comment]
Any progress on this PR? Anything i can do?

**Comment by gerard33** [comment]
Can you add some tests for the new functions so we can make coveralls happy?

Besides that it looks good to me.

**Comment by EddyK69** [comment]
> Can you add some tests for the new functions so we can make coveralls happy?

Done!

**Comment by gerard33** [comment]
Thx, will do a final check asap.

**Review by gerard33 — APPROVED** [review]

**Pull request merged**